### PR TITLE
swarm/storage: Add timeout to netstore private get

### PR DESCRIPTION
**Pull request by nolash**
This PR adds the possibility to set a timeout for a single attempt retrieving chunk through netstore. It is needed for smooth operation of resource update version polling.

### CODE REVIEW

**Review by janos — APPROVED** [review]

**Comment by gbalint** [comment]
Please rebase and resolve the conflicts

File: swarm/storage/netstore.go [hunk 1]
```diff
@@ -121,7 +121,10 @@ func (self *NetStore) Get(key Key) (chunk *Chunk, err error) {
 	}
 }
 
-func (self *NetStore) get(key Key) (chunk *Chunk, err error) {
+func (self *NetStore) get(key Key, timeout time.Duration) (chunk *Chunk, err error) {
+	if timeout == 0 {
```
Comment (nonsense) [inline swarm/storage/netstore.go:125]: This is not obvious behaviour, I think we should be explicit with the params. We are sending `defaultRetrieveTimeout` everywhere this is used, and we are passing 0 at only one place - I think being explicit would be easier to read and reason about.

Comment (nonsense) [reply swarm/storage/netstore.go:125]: I am also not sure why we are using two different timeouts for this. Ultimately this should come from a `context.Context` once we refactor this functionality.

**Review by nonsense — APPROVED** [review]
Added two comments, but feel free to merge as we are going to refactor this anyways in the near future.

File: swarm/storage/netstore.go (hunk 1 above)
Comment (nolash) [reply swarm/storage/netstore.go:125]: It should be `context` I agree.

**Comment by nolash** [comment]
Yeah, let's catch this in refactor. Thanks.

**Pull request merged**